mobile-widgets/qmlmanager.cpp

```diff
@@ -108,6 +109,24 @@ extern "C" int gitProgressCB(const char *text)
 	return 0;
 }
 
+// this callback is used from the uiNotification() function
+// the detour via callback allows us to keep the core code independent from QMLManager
```

Comment:
Not criticizing the code - but that comment is not very convincing. You could achieve the same by defining two different uiNotification functions on desktop and mobile:
```
#ifdef SUBSURFACE_MOBILE
inline void uiNotification(const QString &msg)
{
        showProgress(msg)
}
#else
inline void uiNotification(const QString &msg)
{
}
#endif
```
This way, the compiler *might* remove the whole call on desktio. I believe it won't though because it can't know whether creation/destruction of the QString has side-effects. If you want it to be zero-cost on desktop you would have to define it as a macro.

The callback is needed if you want to change the recipient at runtime. E.g. if such messages might be sent when QMLManager is not yet ready.

Of course, the classical Qt way for that particular use-case would be sending a signal via a class.

Reply:
I'm not worried about the overhead of those calls. Yes, you are 100% correct that I could do all kinds of wonderful things to avoid a few hundred cycles. That is NOT the problem that needs addressing.
I'm happy to rip out the whole notification garbage that is in there. All of it. From everywhere. And create a reasonable way to deal with that and distributing the notifications to the right recipients. Because what we have today is a set of horrible hacks, implemented by different people at different times trying to address exactly their one problem at that time - just like this PR is.

What we are doing today is broken. The user on the mobile app gets no useful feedback and stares at what looks like a hung app for 15 seconds for a fairly modestly sized dive list or 2:30 minutes for the 8000 dive data set (with 98 seconds spent on that `endResetModel()` call).

Please understand that I am NOT saying that there is anything wrong with your code. I am trying to address a serious issue for people actually using the mobile app. I have received half a dozen support requests already where people believe the app has died when in fact it's just processing data before showing the UI.

Reply:
huh?
I was commenting on a non-sequitur code-comment - you went off on a rant about something completely different?

Notably, I'm not interested in cycles here, that was purely an FYI remark. I just pointed out that the comment is not convincing in justifying the complexities of a call-back.

Reply:
Also, thinking about this some more... I'm not convinced that I wouldn't want some form of notification on the desktop as well. Much of the backend code is indeed shared, which is why I have that function pointer mess in the first place.

Reply:
> huh?
> I was commenting on a non-sequitur code-comment - you went off on a rant about something completely different?
> 
> Notably, I'm not interested in cycles here, that was purely an FYI remark. I just pointed out that the comment is not convincing in justifying the complexities of a call-back.

Ignore my rants today, please. I think I said this elsewhere. I am not doing well right now as far as my mental health is concerned.

As for the complexity of a call-back. As a C programmer thinking about the cycles that we take for the callback vs the creation of any one random Qt data structure...
right, you aren't commenting on cycles. so I guess I just don't know what you were really commenting on. Anyway, apologies that my responses are a lot more aggressive than they should be